### ISSUE TITLE
`execute_iter` takes the ownership of passed `PreparedStatement`

### ISSUE THREAD

**Issue by havaker** [scylladb/scylla-rust-driver#266]
In current implementation, `execute_iter` method of `Session` has following signature:
```rust
pub async fn execute_iter(
    &self,
    prepared: impl Into<PreparedStatement>,
    values: impl ValueList,
) -> Result<RowIterator, QueryError> {
```
Executing certain `PreparedStatement` multiple times requires cloning it every time before calling `execute_iter`. This can degrade performance because `PreparedStatement` can be quite large. 

Wrapping prepared statement in `Arc` could be a potential fix:
```rust
pub async fn execute_iter(
    &self,
    prepared: Arc<PreparedStatement>,
    values: impl ValueList,
) -> Result<RowIterator, QueryError> {
```

**Comment by TrueGoric** [comment]
From a performance standpoint I think it would be a welcome addition, `execute_iter` doesn't really need to pass ownership of the `PreparedStatement` since the only place it's modified in this context is in `RowIterator::new_for_prepared_statement` function, where it sets the `page_size` if it isn't set already - I believe this could be easily changed to returning a QueryError on an empty value or making the page size a property of the RowIterator.

**Comment by Lorak-mmk** [comment]
This is mostly addressed by #653 as @wprzytula said, and the rest is covered by issue #811 , so I think this can be closed.
